line_scanner.coffee

```diff
@@ -0,0 +1,92 @@
+tokens = require './tokens'
+
+class LineScanner
+  constructor: (@line) ->
+    @start = 0
+    @position = 0
+    @lineTokens = []
+    # TODO: IMPLEMENT STRING LITERALS 
+    @currentState =
+      multiline:
+        comment: false,
+        string: false
+
+  scan: ->
+    return unless @line
+
+    while @position < @line.length
+
+      @skipSpaces()
+      @skipComments()
+      break unless @position < @line.length
+
+      # continue to next iteration of while loop
+      # once we've been able to extract a token
+      continue if @extractTwoCharacterTokens()
+      continue if @extractOneCharacterTokens()
+      continue if @extractWords()
+      continue if @extractNumericLiterals()
+
+      # return an error if we were not able to 
+      # extract any tokens from the line
+      return {errors: "invalid token bro"}
```

Comment:
obviously we would have better error handling here...
I vote that we create an error module